include/deal.II/lac/block_vector_base.h

```diff
@@ -2264,6 +2262,17 @@ void BlockVectorBase<VectorType>::extract_subvector_to (ForwardIterator
     }
 }
 
+
+template <typename VectorType>
+inline
+bool BlockVectorBase<VectorType>::supports_distributed_data() const
+{
+  // Create a dummy vector of type VectorType to check
+  // supports_distributed_data even if BlockVectorBase contains no block
+  VectorType dummy;
+  return dummy.supports_distributed_data();
```

Comment:
This is pretty much how it works right now. If you try to add two vectors which are derived from ```VectorSpaceVector``` but belong to different derived classes, you will get a runtime error. Nothing forbids to implement a case where you could add two vectors of different types but there is no plan to do so.